plugins/serializers/carbon2/carbon2.go

```diff
@@ -93,6 +94,24 @@ func (s *Serializer) createObject(metric telegraf.Metric) []byte {
 	return m.Bytes()
 }
 
+func (s *Serializer) SetMetricsFormat(f format) {
+	s.metricsFormatLock.Lock()
```

Comment:
are locks needed here?

Reply:
In general yes because we don't know how it will be called.
Perhaps calling `SetMetricsFormat` and `IsMetricsFormatUnset` from different goroutines will never happen but this is just to be sure.
It's not a hot path so my assumption is that it's ok.
Unless there's a specific concern here I'd leave it this way.

Reply:
Unless you've specifically seen race conditions, this should be unnecessary. Plugins have their own dedicated goroutines and the interface functions are never called in parallel by Telegraf. We try to avoid this type of defensive programming, as it tends to be confusing and sets a bad example for other plugins.

Reply:
Got it. Removed.